store/sqlstore/user_store.go

```diff
@@ -1661,3 +1661,164 @@ func applyViewRestrictionsFilter(query sq.SelectBuilder, restrictions *model.Vie
 
 	return resultQuery
 }
+
+func (us SqlUserStore) PromoteGuestToUser(userId string) *model.AppError {
+	transaction, err := us.GetMaster().Begin()
+	if err != nil {
+		return model.NewAppError("SqlUserStore.PromoteGuestToUser", "store.sql_user.promote_guest.open_transaction.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+	defer finalizeTransaction(transaction)
+
+	user, appErr := us.Get(userId)
+	if appErr != nil {
+		return appErr
+	}
+
+	roles := user.GetRoles()
+
+	for idx, role := range roles {
+		if role == "system_guest" {
+			roles[idx] = "system_user"
+		}
+	}
+
+	query := sq.Update("Users").
+		Set("Roles", strings.Join(roles, " ")).
+		Where(sq.Eq{"Id": userId})
+
+	if us.DriverName() == model.DATABASE_DRIVER_POSTGRES {
+		query = query.PlaceholderFormat(sq.Dollar)
+	}
```

Comment:
Doesn't this get done automatically if you use `Where("Users.Id = ?", userId)`?

Reply:
nope, you have to explicitly setup this configuration to make it work with postgres.

Reply:
We're doing it like I suggested here, for example: https://github.com/mattermost/mattermost-server/pull/10701/files#diff-0b7dcefeb566578492d6be5ea6b2453cR898. Or maybe I'm misunderstanding?

Reply:
ok, the difference is you are using `getQueryBuilder`at the beginning, and I'm using directly squirrel library, I think can change that, let me check.

Reply:
Done! :)